index.js

```diff
@@ -3,12 +3,13 @@ const { upload } = require('docker-image-artifact');
 const githubActionIO = require('./actions_io');
 
 const INPUT_IMAGE = 'image';
-
+const RETENTION_DAYS= 'retention_days'
 const OUTPUT_ARTIFACT_NAME = 'artifact_name';
 
 async function runAction(getInput, writeOutput) {
     const imageName = getInput(INPUT_IMAGE, true);
-    const artifactName = await upload(imageName);
+    const retentionDays = getInput(RETENTION_DAYS, false);
+    const artifactName = await upload(imageName, retentionDays);
```

Comment:
`upload` function used here is a wrapper over `upload` function available in `docker-image-artifact` package. For this to work we will have add support for `retentionDays` in the wrapper function. Care to open a PR updating https://github.com/ishworkh/docker-image-artifact/blob/master/src/image_artifact.js#L23 ? :)

Reply:
@ishworkh can you please check whether https://github.com/ishworkh/docker-image-artifact/pull/1 serves the purpose here?

Reply:
https://github.com/ishworkh/docker-image-artifact/pull/1 is merged FYI. 

Lets update here accordingly. Roughly it should be like,

- update package-lock.json `npm update`
- add optional parameter for the action i.e in `action.yaml`
```yaml
inputs:
  image: 
    description: Docker image to upload
    required: true
  retentionDays:
    description: Retention days for the image artifact
    required: false
    default: '0'
```
- update runAction function
```js
const INPUT_RETENTION_DAYS = 'retentionDays'
async function runAction(getInput, writeOutput) {
    const imageName = getInput(INPUT_IMAGE, true);
    const retentionDays = parseInt(getInput(INPUT_RETENTION_DAYS));
    
    const artifactName = await upload(imageName, retentionDays);
    writeOutput(OUTPUT_ARTIFACT_NAME, artifactName);
}
```
- update package.json version to again v1.2.0

Reply:
The `package.json` is not used here atm so it might not matter right now. But the plan is to add an automated github workflow that does the tagging and release based on `package.json` in future.

Reply:
Thanks for the details.